core/src/main/scala/tech/beshu/ror/accesscontrol/domain.scala

```diff
@@ -461,28 +463,47 @@ object domain {
-    val searchAction = Action("indices:data/read/search")
-    val mSearchAction = Action("indices:data/read/msearch")
-    val restoreSnapshotAction = Action("cluster:admin/snapshot/restore")
-    val fieldCapsAction = Action("indices:data/read/field_caps")
-    val asyncSearchAction = Action("indices:data/read/async_search/submit")
-    val rollupSearchAction = Action("indices:data/read/xpack/rollup/search")
-    val searchTemplateAction = Action("indices:data/read/search/template")
-    val putTemplateAction = Action("indices:admin/template/put")
-    val putIndexTemplateAction = Action("indices:admin/index_template/put")
-    val getSettingsAction = Action("indices:monitor/settings/get")
-    val monitorStateAction = Action("cluster:monitor/state")
     // ROR actions
-    val rorUserMetadataAction = Action("cluster:ror/user_metadata/get")
-    val rorConfigAction = Action("cluster:ror/config/manage")
-    val rorTestConfigAction = Action("cluster:ror/testconfig/manage")
-    val rorAuthMockAction = Action("cluster:ror/authmock/manage")
-    val rorAuditEventAction = Action("cluster:ror/audit_event/put")
-    val rorOldConfigAction = Action("cluster:ror/config/refreshsettings")
+    val rorUserMetadataAction: Action = rorAction("user_metadata/get")
+    val rorConfigAction: Action = rorAction("config/manage")
+    val rorTestConfigAction: Action = rorAction("testconfig/manage")
+    val rorAuthMockAction: Action = rorAction("authmock/manage")
+    val rorAuditEventAction: Action = rorAction("audit_event/put")
+    val rorOldConfigAction: Action = rorAction("config/refreshsettings")
+
+    private val searchAction: Action = new Action("indices:data/read/search")
+    private val mSearchAction: Action = new Action("indices:data/read/msearch")
+    private val fieldCapsAction: Action = new Action("indices:data/read/field_caps")
+    private val asyncSearchAction: Action = new Action("indices:data/read/async_search/submit")
+    private val rollupSearchAction: Action = new Action("indices:data/read/xpack/rollup/search")
+    private val searchTemplateAction: Action = new Action("indices:data/read/search/template")
+    private val putTemplateAction: Action = new Action("indices:admin/template/put")
+    private val putIndexTemplateAction: Action = new Action("indices:admin/index_template/put")
+    private val getSettingsAction: Action = new Action("indices:monitor/settings/get")
+    private val monitorStateAction: Action = new Action("cluster:monitor/state")
+
+    private val rorActionByOutdatedName: Map[String, Action] = Map(
+      "cluster:ror/user_metadata/get" -> rorUserMetadataAction,
+      "cluster:ror/config/manage" -> rorConfigAction,
+      "cluster:ror/testconfig/manage" -> rorTestConfigAction,
+      "cluster:ror/authmock/manage" -> rorAuthMockAction,
+      "cluster:ror/audit_event/put" -> rorAuditEventAction,
+      "cluster:ror/config/refreshsettings" -> rorOldConfigAction
+    )
+
+    def apply(value: String): Action = {
```

Comment:
I'm not sure if this is the right approach. We will see when you write the test I mentioned. 

---
I was rather thinking that we should do it like this:
```
sealed trait Action
object Action {
  final case class EsAction(name) extends Action
  sealed trait RorAction extends Action
  object RorAction {
    case object RorUserMetadataAction extends RorAction
    ...
  }
  
  def from(value: String): Action = {
    // two strings can lead to create RorUserMetadataAction
  }
}